.circleci/config.yml

```diff
@@ -12,76 +12,183 @@ version: 2.1
+  compile:
+    parameters:
+      compile-mode:
+        type: string
+        default: production
     steps:
-      - checkout
-      - restore_cache: *restore_cache
-      - run: npm install
       - run:
-          name: 'Compile in production mode'
+          name: 'Compile in << parameters.compile-mode >> mode'
           environment:
-            - NODE_ENV: production
+            - NODE_ENV: << parameters.compile-mode >>
           command: |
             npm run compile
+
+  gh-config:
+    parameters:
+      cache_timeout:
+        description: 'Cache timeout in seconds'
+        type: integer
+        default: 120
+      gh_email:
+        description: 'GitHub email'
+        type: string
+        default: $GH_EMAIL
+      gh_username:
+        description: 'GitHub username'
+        type: string
+        default: 'Release Bot'
+    steps:
+      - run:
+          name: 'Configuring GitHub'
+          command: |
+            git config credential.helper 'cache --timeout=<< parameters.cache_timeout >>'
+            git config user.email "<< parameters.gh_email >>"
+            git config user.name "<< parameters.gh_username >>"
+
+  gus-prepare-environment-variables:
+    parameters:
+      package-json-path:
+        type: string
+        default: './packages/salesforcedx-vscode/package.json'
+      release-tag-name:
+        type: string
+        default: 'offcore.tooling'
+    steps:
+      - run:
+          name: 'Prepare environment variables for GUS Change Case Management'
+          command: |
+            export GUS_BUILD="$(node -pe "require('<< parameters.package-json-path >>').version")"
+            echo "export RELEASE_VERSION=$GUS_BUILD" >> $BASH_ENV
+            echo "export GUS_RELEASE=<< parameters.release-tag-name >>.$GUS_BUILD" >> $BASH_ENV
+            source $BASH_ENV
+            echo "RELEASE_VERSION is: ${RELEASE_VERSION}"
+            echo "GUS_RELEASE is: ${GUS_RELEASE}"
+
+  install:
+    description: 'Installing dependencies'
+    steps:
+      - run: npm install
+      - run: npm --version
+      - run: node --version
+
+  package-extensions:
+    steps:
       - run:
           name: 'Packaging extensions'
           command: npm run vscode:package
+
+  restore_dependency_cache:
+    # This logic cannot be done within the node orb currently. This is due to the fact that we use
+    # the package.json instead of the package-lock.json for our checksum.
+    parameters:
+      base-cache-path:
+        type: string
+        default: 'dependency-cache-v2-'
+    steps:
+      - restore_cache:
+          keys:
+            - << parameters.base-cache-path >>{{ checksum "package.json" }}
+            # if cache for exact version of 'package.json' is not present, then load the most recent cache
+            - << parameters.base-cache-path >>
+
+  save_dependency_cache:
+    parameters:
+      base-cache-path:
+        type: string
+        default: 'dependency-cache-v2-'
```

Comment:
Let's update the default to `dependency-cache-v3-`, since we updated this last week because we wanted to flush the cache. Which going forward will be easier now that it's a param.